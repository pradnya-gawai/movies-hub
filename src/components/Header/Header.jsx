import React from "react";
import "./Header.css";

function Header() {
  return (
    <div>
      <span className="header" data-testid="header">Movies Hub </span>
    </div>
  );
}

export default Header;
