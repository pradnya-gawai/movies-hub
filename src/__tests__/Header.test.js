import Header from "../components/Header/Header";
import { render } from "@testing-library/react";
import { unmountComponentAtNode } from "react-dom";
import { act } from "react-test-renderer";

let container = null;
beforeEach(() => {
  // setup a DOM element as a render target
  container = document.createElement("div");
  document.body.appendChild(container);
});

afterEach(() => {
  // cleanup on exiting
  unmountComponentAtNode(container);
  container.remove();
  container = null;
});

describe("Header", () => {
  test("should render  header correctly", () => {
    const wrapper = shallow(<Header />);
    expect(wrapper).toMatchSnapshot();
  });
  
  test("Renders's the header and required inputs", () => {
    const { getByText, getByTestId } = render(<Header />);
    expect(getByTestId("header")).toBeTruthy();
    expect(getByText("Movies Hub")).toBeTruthy();
    // matching textContent
    expect(getByTestId("header").textContent).toMatch("Movies Hub");
  });
});
