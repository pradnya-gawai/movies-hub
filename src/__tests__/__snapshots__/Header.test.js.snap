// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header should render  header correctly 1`] = `
<div>
  <span
    className="header"
    data-testid="header"
  >
    Movies Hub 
  </span>
</div>
`;
